### PR TITLE
prevent dollars to be passed to another function

### PR DESCRIPTION
**Pull request by jantimon**
there was a bug which caused invalid prop forwarding

this pr adds the following test and ensures that it passes:

```ts
it("should filter out properties starting with $ when passing to custom", () => {
  let forwardedProps = null;
  const Component = ({ className, style, ...props}) => {
    forwardedProps = props;
    return null;
  };
  const StyledComponent = styled(Component)``;
  const { container } = render(<StyledComponent $forwardedProp="notForwarded" />);

  expect(forwardedProps).toEqual({});
});
```

### CODE REVIEW

**Review by rvetere — APPROVED** [review]
looks good to me

File: packages/next-yak/runtime/__tests__/attrs.test.tsx [hunk 1]
```diff
@@ -530,11 +517,8 @@ it("should pass theme if theme is overwritten", () => {
     ).toJSON(),
   ).toMatchInlineSnapshot(`
     <pre
-      $__attrs={true}
       className=""
       style={{}}
-    >
-      {"color":"blue"}
-    </pre>
+    />
```
Comment (Mad-Kat) [inline packages/next-yak/runtime/__tests__/attrs.test.tsx:522]: There should be the theme prop passed, shouldn't it?

File: packages/next-yak/runtime/__tests__/styled.test.tsx [hunk 2]
```diff
@@ -302,9 +329,7 @@ it("should not remove theme if theme is passed to wrapped element", () => {
     <div>
       <pre
         class="test-wrapper test"
-      >
-        {"anything":"test"}
-      </pre>
+      />
```
Comment (Mad-Kat) [inline packages/next-yak/runtime/__tests__/styled.test.tsx:332]: Same as above

**Review by Mad-Kat — CHANGES_REQUESTED** [review]
Generally the change looks good and I don't know where the -24% execution time comes in, but I like it 😄 

I think the approach of not forwarding the props if it's not a yak-component is generally good, but we have to think about the case with the theme. 

What do you think?

File: packages/next-yak/runtime/__tests__/styled.test.tsx [hunk 3]
```diff
@@ -302,9 +329,7 @@ it("should not remove theme if theme is passed to wrapped element", () => {
     <div>
       <pre
         class="test-wrapper test"
-      >
```
Comment (jantimon) [inline packages/next-yak/runtime/__tests__/styled.test.tsx:305]: this is probably wrong

**Comment by jantimon** [comment]
> Generally the change looks good and I don't know where the -24% execution time comes in, but I like it 😄
> 
> I think the approach of not forwarding the props if it's not a yak-component is generally good, but we have to think about the case with the theme.
> 
> What do you think?

Maybe I found a fix - lets see if the pipeline is happy

**Comment by jantimon** [comment]
The performance gain is that it now finally detects nested yak components and calls them directly without telling react 🤣

**Review by Mad-Kat — APPROVED** [review]
Good job. Really nice change

**Pull request merged**